### ISSUE TITLE
build fail [SOLVED]

### ISSUE THREAD

**Issue by davephillips** [BespokeSynth/BespokeSynth#10]
Hmm, maybe a lib isn't getting linked ?

```
Linking BespokeSynth - App
build/intermediate/Release/Push2-Usb-Communicator_2ebfaa0a.o: In function `ableton::UsbCommunicator::PollUsbForEvents()':
Push2-Usb-Communicator.cpp:(.text+0x3d): undefined reference to `libusb_handle_events_timeout_completed'
build/intermediate/Release/Push2-Usb-Communicator_2ebfaa0a.o: In function `ableton::UsbCommunicator::sendNextSlice(libusb_transfer*)':
Push2-Usb-Communicator.cpp:(.text+0x3f6): undefined reference to `libusb_submit_transfer'
Push2-Usb-Communicator.cpp:(.text+0x452): undefined reference to `libusb_submit_transfer'
build/intermediate/Release/Push2-Usb-Communicator_2ebfaa0a.o: In function `ableton::UsbCommunicator::startSending()':
Push2-Usb-Communicator.cpp:(.text+0x6b1): undefined reference to `libusb_alloc_transfer'
Push2-Usb-Communicator.cpp:(.text+0x710): undefined reference to `libusb_alloc_transfer'
build/intermediate/Release/Push2-Usb-Communicator_2ebfaa0a.o: In function `ableton::UsbCommunicator::Init(unsigned short const*)':
Push2-Usb-Communicator.cpp:(.text+0x810): undefined reference to `libusb_init'
Push2-Usb-Communicator.cpp:(.text+0x824): undefined reference to `libusb_set_debug'
Push2-Usb-Communicator.cpp:(.text+0x830): undefined reference to `libusb_get_device_list'
Push2-Usb-Communicator.cpp:(.text+0x8c5): undefined reference to `libusb_get_device_descriptor'
Push2-Usb-Communicator.cpp:(.text+0x8fd): undefined reference to `libusb_open'
Push2-Usb-Communicator.cpp:(.text+0x911): undefined reference to `libusb_claim_interface'
Push2-Usb-Communicator.cpp:(.text+0x941): undefined reference to `libusb_close'
Push2-Usb-Communicator.cpp:(.text+0xbe7): undefined reference to `libusb_free_device_list'
collect2: error: ld returned 1 exit status
Makefile:385: recipe for target 'build/BespokeSynth' failed
make: *** [build/BespokeSynth] Error 1

```

**Comment by awwbees** [comment]
I've added a dependency on libusb, maybe on linux it needs to be installed manually with your package manager?

________________________________

**Comment by davephillips** [comment]
Okay, that was a bit of an ordeal. It seems that Ubuntu doesn't install libusb in the obvious places, so here's what's needed for building on Ubuntu 18.04. These additions must be made to the BespokeSynth.jucer file:

Add this to External Libraries To Link:

`usb-1.0`

and this to Extra Library Search Paths in both the Debug and Release sections:

`/usr/lib/x86_64-linux-gnu/`

BespokeSynth will then build and run. However, selecting the push2control from the Instruments crashes Bespoke with this error:

```
Shader shader/vert error:

Segmentation fault (core dumped)
```

**Comment by awwbees** [comment]
thank you for figuring that out! fix submitted in 004c95f

the crash you were getting after fixing it was because you didn't have Ableton Push 2 hardware connected, and that has also been fixed to be handled more gracefully.

I haven't yet tested push2control on linux, because I'm running in a VM so it doesn't show up as a USB device for me. I'll have to do a real linux install so I can test it. it might already work!
